client/packages/system/src/Patient/Components/PatientSearchInput/PatientSearchInput.tsx

```diff
@@ -1,42 +1,42 @@
 import React, { FC } from 'react';
-import { Autocomplete, useBufferState } from '@openmsupply-client/common';
+import { Autocomplete } from '@openmsupply-client/common';
 import {
   NameSearchInputProps,
-  basicFilterOptions,
+  SearchInputPatient,
   filterByNameAndCode,
 } from '../../utils';
-import { usePatient } from '../../api';
 import { getPatientOptionRenderer } from '../PatientOptionRenderer';
+import { searchPatient } from '../utils';
 
 export const PatientSearchInput: FC<NameSearchInputProps> = ({
   onChange,
   width = 250,
   value,
   disabled = false,
 }) => {
-  const { data, isLoading } = usePatient.document.list({
-    sortBy: { key: 'name', direction: 'asc' },
-  });
-  const [buffer, setBuffer] = useBufferState(value);
   const PatientOptionRenderer = getPatientOptionRenderer();
+  const { debouncedOnChange, isLoading, patients, setSearchText } =
+    searchPatient();
 
   return (
     <Autocomplete
+      options={patients ?? []}
       disabled={disabled}
       clearable={false}
-      value={buffer && { ...buffer, label: buffer.name }}
-      filterOptionConfig={basicFilterOptions}
-      filterOptions={filterByNameAndCode}
       loading={isLoading}
+      onInputChange={(_, value) => {
+        debouncedOnChange(value);
+        setSearchText(value);
+      }}
       onChange={(_, name) => {
-        setBuffer(name);
-        name && onChange(name);
+        if (name && !(name instanceof Array)) onChange(name);
       }}
-      options={data?.nodes ?? []}
       renderOption={PatientOptionRenderer}
+      getOptionLabel={(option: SearchInputPatient) => option.name}
       width={`${width}px`}
       popperMinWidth={width}
-      isOptionEqualToValue={(option, value) => option?.id === value?.id}
+      defaultValue={value && { ...value, label: value.name }}
```

Comment:
Used this to show the patient name in prescription view... value was bugging for me ;-;